plugin/src/main/java/org/opensearch/ml/model/MLModelAndNodeManager.java

```diff
@@ -0,0 +1,294 @@
+ */
+@Log4j2
+public class MLModelAndNodeManager {
+    private final Client client;
+    private final ClusterService clusterService;
+    private final ThreadPool threadPool;
+    private final NamedXContentRegistry xContentRegistry;
+    private final ModelHelper modelHelper;
+    private final DiscoveryNodeHelper nodeHelper;
+    private final MLModelCacheHelper modelCacheHelper;
+    private final MLStats mlStats;
+    private final MLCircuitBreakerService mlCircuitBreakerService;
+    private final MLIndicesHandler mlIndicesHandler;
+    private final MLTaskManager mlTaskManager;
+    private final MLEngine mlEngine;
+    private volatile Boolean enableAutoReLoadModel;
+
+    public MLModelAndNodeManager(
+        ClusterService clusterService,
+        Client client,
+        ThreadPool threadPool,
+        NamedXContentRegistry xContentRegistry,
+        ModelHelper modelHelper,
+        DiscoveryNodeHelper nodeHelper,
+        Settings settings,
+        MLStats mlStats,
+        MLCircuitBreakerService mlCircuitBreakerService,
+        MLIndicesHandler mlIndicesHandler,
+        MLTaskManager mlTaskManager,
+        MLModelCacheHelper modelCacheHelper,
+        MLEngine mlEngine
+    ) {
+        this.clusterService = clusterService;
+        this.client = client;
+        this.threadPool = threadPool;
+        this.xContentRegistry = xContentRegistry;
+        this.modelHelper = modelHelper;
+        this.nodeHelper = nodeHelper;
+        this.modelCacheHelper = modelCacheHelper;
+        this.mlStats = mlStats;
+        this.mlCircuitBreakerService = mlCircuitBreakerService;
+        this.mlIndicesHandler = mlIndicesHandler;
+        this.mlTaskManager = mlTaskManager;
+        this.mlEngine = mlEngine;
+
+        enableAutoReLoadModel = ML_COMMONS_MODEL_AUTO_RELOAD_ENABLE.get(settings);
+        clusterService
+            .getClusterSettings()
+            .addSettingsUpdateConsumer(ML_COMMONS_MODEL_AUTO_RELOAD_ENABLE, it -> enableAutoReLoadModel = it);
+    }
+
+    public void autoReLoadModel() {
+        log.debug("enableAutoReLoadModel {} ", enableAutoReLoadModel);
+
+        // if we don't need to reload automatically, just return without doing anything
+        if (!enableAutoReLoadModel) {
+            return;
+        }
+
+        // At opensearch startup, get local node id, if not ml node,we ignored, just return without doing anything
+        String localNodeId = clusterService.localNode().getId();
+        DiscoveryNode node = nodeHelper.getNode(localNodeId);
+        if (!MLNodeUtils.isMLNode(node)) {
+            return;
+        }
+
+        // According to the node id to query the number of retries, if more than 2 (the maximum number of retries), we do not need to retry,
+        // that the number of unsuccessful reload has reached the maximum number of times, do not need to reload
+        Integer reTryTimes = 0;
+        if (isExistedIndex(ML_MODEL_RELOAD_INDEX)) {
```

Comment:
Do we need a specific check of existing? Since getRetryTimes also query on this index, we can use the http code like `404` to figure out if the index exists or not.

Reply:
we need when the first time to reload model. at that time, the `ML_MODEL_RELOAD_INDEX` didn't exist,so wee haven't saved reTryTimes before that time.